### PR TITLE
Add NFData Graph instance

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: algebraic-graphs.cabal [hunk 1]
```diff
@@ -69,7 +69,8 @@ library
                         Algebra.Graph.Relation.Transitive
     build-depends:      array      >= 0.5 && < 0.8,
                         base       >= 4.9 && < 5,
-                        containers >= 0.5 && < 0.8
+                        containers >= 0.5 && < 0.8,
+                        deepseq    >= 1.4.2.0 && <1.5
```
Comment (phadej) [inline algebraic-graphs.cabal]: that's the version distributed with GHC-8.0

File: src/Algebra/Graph.hs [hunk 2]
```diff
@@ -47,6 +47,7 @@ module Algebra.Graph (
   ) where
 
 import Control.Applicative (Alternative, (<|>))
+import Control.Deepseq (NFData (..))
```
Comment (snowleopard) [inline src/Algebra/Graph.hs]: Looks like this should be `DeepSeq` instead -- see AppVeyor build log:

https://travis-ci.org/snowleopard/alga/jobs/300622428

File: algebraic-graphs.cabal (hunk 1 above)
Comment (snowleopard) [reply algebraic-graphs.cabal]: I'll see if Alga works with earlier versions of GHC and might move this lower bound to allow earlier GHC versions.

**Comment by snowleopard** [comment]
@phadej Thanks you, merged!

Will do the release soon.

**Pull request merged**